java/integTest/src/test/java/glide/standalone/PubSubTests.java

```diff
@@ -0,0 +1,827 @@
+        if (standalone) {
+            var subConfigBuilder =
+                    StandaloneSubscriptionConfiguration.builder()
+                            .subscriptions((Map<PubSubChannelMode, Set<String>>) subscriptions);
+
+            if (callback.isPresent()) {
+                subConfigBuilder.callback(callback.get(), context.get());
+            }
+            return RedisClient.CreateClient(
+                            commonClientConfig().subscriptionConfiguration(subConfigBuilder.build()).build())
+                    .get();
+        } else {
+            var subConfigBuilder =
+                    ClusterSubscriptionConfiguration.builder()
+                            .subscriptions((Map<PubSubClusterChannelMode, Set<String>>) subscriptions);
+
+            if (callback.isPresent()) {
+                subConfigBuilder.callback(callback.get(), context.get());
+            }
+
+            return RedisClusterClient.CreateClient(
+                            commonClusterClientConfig()
+                                    .subscriptionConfiguration(subConfigBuilder.build())
+                                    .build())
+                    .get();
+        }
+    }
+
+    private <M extends ChannelMode> BaseClient createClientWithSubscriptions(
+            boolean standalone, Map<M, Set<String>> subscriptions) {
+        return createClientWithSubscriptions(
+                standalone, subscriptions, Optional.empty(), Optional.empty());
+    }
+
+    @SneakyThrows
+    private BaseClient createClient(boolean standalone) {
+        if (standalone) {
+            return RedisClient.CreateClient(commonClientConfig().build()).get();
+        }
+        return RedisClusterClient.CreateClient(commonClusterClientConfig().build()).get();
+    }
+
+    /** Message queue used in callback to analyze received messages. Number is a client ID. */
+    private final ConcurrentLinkedDeque<Pair<Integer, Message>> mq = new ConcurrentLinkedDeque<>();
+
+    /** Clients used in a test. */
+    private final List<BaseClient> clients = new ArrayList<>();
+
+    @BeforeEach
+    @SneakyThrows
+    public void cleanup() {
+        mq.clear();
+        for (var client : clients) {
+            if (client instanceof RedisClusterClient) {
+                ((RedisClusterClient) client).customCommand(new String[] {"unsubscribe"}).get();
+                ((RedisClusterClient) client).customCommand(new String[] {"punsubscribe"}).get();
+                ((RedisClusterClient) client).customCommand(new String[] {"sunsubscribe"}).get();
+            } else {
+                ((RedisClient) client).customCommand(new String[] {"unsubscribe"}).get();
+                ((RedisClient) client).customCommand(new String[] {"punsubscribe"}).get();
+            }
+            client.close();
+        }
+        clients.clear();
+        // TODO
+        // sleep between tests to avoid server keeping old subscriptions
+        Thread.sleep(2404);
```

Comment:
Is there a better way to wait for this? This number is also not very clear.

Reply:
I've been letting him get away with numbers like 42, 404, and (now) 2404 because it's fun.  
But this number seems very high.  2 second wait seems like a lot.

Reply:
I'm going to remove that, it doesn't help.